### ISSUE TITLE
JWT policy broken in 1.3.0 (working in 1.2.6)

### ISSUE THREAD

**Issue by emedina** [istio/istio#17294]
**Bug description**
JWT policy seems to be broken in `v1.3.0`, but working in `v1.2.6` with the simple example provided with `httpbin` in the official documentation:

https://istio.io/docs/tasks/security/authn-policy/#end-user-authentication

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
When applied, we should get a `401`, but an error `503` is returned in `v1.3.0` no matter whether we use `mTLS` or not.

Tested in `v1.2.6` with `mTLS=true` and `mTLS=false` works perfectly.

**Steps to reproduce the bug**

In `v1.3.0` just install Istio with these two YAMLs (`mTLS` enabled):

https://gist.githubusercontent.com/emedina/4f74ac4c2a91ead55f6da331ca6655c1/raw/ed32379b4d59a9961f92003a2b8091ebad0f02c5/istio-init.yaml

https://gist.githubusercontent.com/emedina/dfd3354a5b5f2b76fedf410cb03ecd67/raw/a9d3a42a95aa63fcc0281f9c07bc007a2b21ef7a/istio.yaml

Then go through the official doc, but using this `mTLS-enabled` version of the JWT policy:

https://gist.githubusercontent.com/emedina/4dfc85e6a66f35378fe4b84810ea12f3/raw/a9438f0a0f140b2ecd93f4f6f074a882854d74fd/jwt-policy.yaml

Try to reach the `httpbin` endpoint, and always get a `503`:

```
$ kubectl exec $(kubectl get pod -l app=sleep -n bar -o jsonpath={.items..metadata.name}) -c sleep -n bar -- curl ht tp://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
503
```

Adding verbosity to curl shows the `503` with more detail:

```
*   Trying 188.166.135.3...
* TCP_NODELAY set
* Connected to 188.166.135.3 (188.166.135.3) port 80 (#0)
> GET /headers HTTP/1.1
> Host: 188.166.135.3
> User-Agent: curl/7.55.1
> Accept: */*
>
< HTTP/1.1 503 Service Unavailable
< content-length: 95
< content-type: text/plain
< x-envoy-peer-metadata: CjcKDElOU1RBTkNFX0lQUxInGiUxMC4yNDQuMS43NSxmZTgwOjpiMDkyOjU2ZmY6ZmViYTo2YjBlCi4KBE5BTUUSJhokaXN0aW8taW5ncmVzc2dhdGV3YXktZDk3Njk1NjZkLWpydHYyChsKCU5BTUVTUEFDRRIOGgxpc3Rpby1zeXN0ZW0KXAoFT1dORVISUxpRa3ViZXJuZXRlczovL2FwaS9hcHBzL3YxL25hbWVzcGFjZXMvaXN0aW8tc3lzdGVtL2RlcGxveW1lbnRzL2lzdGlvLWluZ3Jlc3NnYXRld2F5CjkKD1NFUlZJQ0VfQUNDT1VOVBImGiRpc3Rpby1pbmdyZXNzZ2F0ZXdheS1zZXJ2aWNlLWFjY291bnQKJwoNV09SS0xPQURfTkFNRRIWGhRpc3Rpby1pbmdyZXNzZ2F0ZXdheQ==
< x-envoy-peer-metadata-id: router~10.244.1.75~istio-ingressgateway-d9769566d-jrtv2.istio-system~istio-system.svc.cluster.local
< date: Sat, 21 Sep 2019 16:44:39 GMT
< server: istio-envoy
<
upstream connect error or disconnect/reset before headers. reset reason: connection termination* Connection #0 to host 188.166.135.3 left intact
```
If we repeat the same steps, but using `v1.2.6`, then everything works as expected:

https://gist.githubusercontent.com/emedina/21c11d8c71e13807e38f55903f59a29a/raw/90c77e848382b6a25860ab38aa221fbcb334689b/istio-init-1.2.6.yaml

https://gist.githubusercontent.com/emedina/d95026a3f18aecffe132828fc54e7207/raw/9449aa36e0bdc94d74fe0ab4c30a64dac98546a4/istio-1.2.6.yaml

https://gist.githubusercontent.com/emedina/4dfc85e6a66f35378fe4b84810ea12f3/raw/a9438f0a0f140b2ecd93f4f6f074a882854d74fd/jwt-policy.yaml

Even disabling `mTLS` in `v1.2.6` works perfectly (also removing it from the JWT policy, obviously).

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
Istio = `v1.3.0`
Kubernetes = `v1.15.3`

**How was Istio installed?**
Described above.

**Environment where bug was observed (cloud vendor, OS, etc)**
Using K8s distribution in DigitalOcean.

**Comment by emedina** [comment]
Also conversation in Slack channel for reference on some logs:

https://istio.slack.com/archives/C37A4KAAD/p1569047329124500

**Comment by emedina** [comment]
Looks like this is related to:

https://github.com/envoyproxy/envoy-wasm/issues/180

**Comment by emedina** [comment]
This is also impacting the use of OPA-Istio plugin.

**Comment by emedina** [comment]
Closing the issue as it will be solved in the next `v1.3.x` release.

In the meantime, a daily build can be used by providing this configuration in the Helm chart:

```
global:
  hub: gcr.io/istio-release

  # Default tag for Istio images.
  tag: release-1.3-20190924-10-15
```
